File: frb_example/pure_dart/lib/src/rust/api/external_type_in_crate.dart
// ignore_for_file: invalid_use_of_internal_member, unused_import

import '../auxiliary/new_module_system/sub_module.dart';
import '../auxiliary/old_module_system/sub_module.dart';
import '../auxiliary/sample_types.dart';
import '../frb_generated.dart';
import 'package:flutter_rust_bridge/flutter_rust_bridge_for_generated.dart';

Future<bool> useImportedStruct({required MyStruct myStruct, dynamic hint}) =>
    RustLib.instance.api.useImportedStruct(myStruct: myStruct, hint: hint);

Future<bool> useImportedEnum({required MyEnum myEnum, dynamic hint}) =>
    RustLib.instance.api.useImportedEnum(myEnum: myEnum, hint: hint);

Future<OldSimpleStruct> callOldModuleSystem({dynamic hint}) =>
    RustLib.instance.api.callOldModuleSystem(hint: hint);

Future<NewSimpleStruct> callNewModuleSystem({dynamic hint}) =>
    RustLib.instance.api.callNewModuleSystem(hint: hint);

File: frb_example/pure_dart/rust/src/api/external_type_in_crate.rs
use crate::auxiliary::new_module_system::{use_new_module_system, NewSimpleStruct};
use crate::auxiliary::old_module_system::{use_old_module_system, OldSimpleStruct};
use crate::auxiliary::sample_types::{MyEnum, MyStruct};

// Function that uses imported struct (from within this crate)
pub fn use_imported_struct_twin_normal(my_struct: MyStruct) -> bool {
    my_struct.content
}

// Function that uses imported enum (from within this crate)
pub fn use_imported_enum_twin_normal(my_enum: MyEnum) -> bool {
    match my_enum {
        MyEnum::False => false,
        MyEnum::True => true,
    }
}

pub fn call_old_module_system_twin_normal() -> OldSimpleStruct {
    use_old_module_system(2)
}

pub fn call_new_module_system_twin_normal() -> NewSimpleStruct {
    use_new_module_system(1)
}
